rs/rosetta-api/icrc1/ledger/tests/tests.rs

```diff
@@ -382,16 +392,28 @@ fn test_block_transformation() {
 
 #[cfg_attr(feature = "u256-tokens", ignore)]
 #[test]
-fn icrc1_test_approval_upgrade() {
-    ic_icrc1_ledger_sm_tests::icrc1_test_approval_upgrade(
+fn icrc1_test_upgrade_serialization() {
+    let minter = Arc::new(minter_identity());
+    let builder = LedgerInitArgsBuilder::with_symbol_and_name(TOKEN_SYMBOL, TOKEN_NAME)
+        .with_minting_account(minter.sender().unwrap())
+        .with_transfer_fee(FEE);
+    let init_args = Encode!(&LedgerArgument::Init(builder.build())).unwrap();
+    let upgrade_args = Encode!(&LedgerArgument::Upgrade(None)).unwrap();
+    ic_icrc1_ledger_sm_tests::test_upgrade_serialization(
         ledger_mainnet_wasm(),
```

Comment:
IIUC, instead of calling `ledger_mainnet_wasm()` here, we could call either the `u64` or `u256` version, depending on if the `u256-tokens` feature is defined or not:
```
    #[cfg(not(feature = "u256-tokens"))]
    let ledger_mainnet_wasm = ledger_mainnet_u64_wasm();
    #[cfg(feature = "u256-tokens")]
    let ledger_mainnet_wasm = ledger_mainnet_u256_wasm();
```
That way, we could also remove `#[cfg_attr(feature = "u256-tokens", ignore)]` from the test, allowing us to test upgrades and downgrades for both the `u64` and `u256` ledgers.

Reply:
(Maybe a better approach would be to make the `ledger_mainnet_wasm` return either the `u64` or `u256` mainnet wasm depending on the feature. That method has multiple callers though, and I didn't try it out, so making that change could be considered for another ticket.)

Reply:
Thanks, done!

Reply:
Done!